src/command_modules/azure-cli-ml/azure/cli/command_modules/ml/service/realtime.py

```diff
@@ -0,0 +1,973 @@
+
+    if verbose:
+        print(code)
+    az_container_name = 'amlbdpackages'
+    az_blob_name = str(uuid.uuid4()) + '.py'
+    bbs = BlockBlobService(account_name=context.az_account_name,
+                           account_key=context.az_account_key)
+    bbs.create_container(az_container_name)
+    bbs.create_blob_from_text(az_container_name, az_blob_name, code,
+                              content_settings=ContentSettings(content_type='application/text'))
+    blob_sas = bbs.generate_blob_shared_access_signature(
+        az_container_name,
+        az_blob_name,
+        BlobPermissions.READ,
+        datetime.utcnow() + timedelta(days=30))
+    package_location = 'http://{}.blob.core.windows.net/{}/{}?{}'.format(context.az_account_name,
+                                                                         az_container_name, az_blob_name, blob_sas)
+
+    if verbose:
+        print("Package uploaded to " + package_location)
+
+    for asset in json_payload['properties']['assets']:
+        if asset['id'] == 'driver_package_asset':
+            if verbose:
+                print("Current driver location:", str(asset['location']))
+                print("Replacing with:", package_location)
+            asset['location'] = package_location
+
+    # modify json payload to set ACR credentials
+    if verbose:
+        print("Current ACR creds in payload:")
+        print('location:', json_payload['properties']['registryInfo']['location'])
+        print('user:', json_payload['properties']['registryInfo']['user'])
+        print('password:', json_payload['properties']['registryInfo']['password'])
+
+    json_payload['properties']['registryInfo']['location'] = context.acr_home
+    json_payload['properties']['registryInfo']['user'] = context.acr_user
+    json_payload['properties']['registryInfo']['password'] = context.acr_pw
+
+    if verbose:
+        print("New ACR creds in payload:")
+        print('location:', json_payload['properties']['registryInfo']['location'])
+        print('user:', json_payload['properties']['registryInfo']['user'])
+        print('password:', json_payload['properties']['registryInfo']['password'])
+
+    # call ICE with payload to create docker image
+
+    # Set base ICE URL
+    if custom_ice_url is not '':
+        base_ice_url = custom_ice_url
+        if base_ice_url.endswith('/'):
+            base_ice_url = base_ice_url[:-1]
+    else:
+        base_ice_url = 'https://amlacsagent.azureml-int.net'
+
+    create_url = base_ice_url + '/images/' + service_name
+    get_url = base_ice_url + '/jobs'
+
+    headers = {'Content-Type': 'application/json', 'User-Agent': 'aml-cli-preview-060317'}
+
+    image = ''
+    try:
+        ice_put_result = requests.put(
+            create_url, headers=headers, data=json.dumps(json_payload), timeout=ice_connection_timeout)
+    except (requests.ConnectionError, requests.ConnectTimeout, requests.exceptions.ReadTimeout):
```

Comment:
requests.exceptions.ConnectTimeout, but can remove because it inherits from ConnectionError